pytorch_lightning/loops/epoch/training_epoch_loop.py

```diff
@@ -180,11 +180,10 @@ def advance(self, data_fetcher: AbstractDataFetcher) -> None:  # type: ignore[ov
 
             # hook
             self.trainer._call_callback_hooks("on_train_batch_start", batch, batch_idx, **extra_kwargs)
-            model_response = self.trainer._call_lightning_module_hook(
+            response = self.trainer._call_lightning_module_hook(
                 "on_train_batch_start", batch, batch_idx, **extra_kwargs
             )
-            ttp_response = self.trainer._call_ttp_hook("on_train_batch_start", batch, batch_idx, **extra_kwargs)
-            response = ttp_response if model_response is None else model_response
+            self.trainer._call_ttp_hook("on_train_batch_start", batch, batch_idx, **extra_kwargs)
```

Comment:
`_call_ttp_hook` is not directly annotated as returning None, nor is it supposed to - it is just missing the return type annotation. The reason it is missing the return type annotation is because it previously had `-> Optional[Any]` as return type but this caused a mypy error:
`pytorch_lightning/loops/optimization/optimizer_loop.py:439: error: Argument 1 to "_extract_hiddens" has incompatible type "Optional[Any]"; expected "Union[Tensor, Dict[str, Any]]"  [arg-type]`

so I removed it. But in order to maintain the current behavior of `call_hook`, ttp hooks _should_ be allowed to return something. see previous behavior:
https://github.com/PyTorchLightning/pytorch-lightning/blob/a28b4cd0c0bba30c21cae571e650877f66cf5588/pytorch_lightning/trainer/trainer.py#L1482-L1488

Reply:
Also from the original motivation in https://github.com/PyTorchLightning/pytorch-lightning/issues/8506, don't we want to have the flexibility to return something from ttp hooks?

Reply:
`_call_ttp_hook` can return something at the moment. But the concrete `on_train_batch_start` hook doesn't.

```python
$ grep -iIrn "on_train_batch_start" pytorch_lightning/plugins/training_type/
pytorch_lightning/plugins/training_type//ipu.py:325:    def on_train_batch_start(self, batch: Any, batch_idx: int, dataloader_idx: int = 0) -> None:
pytorch_lightning/plugins/training_type//training_type_plugin.py:456:    def on_train_batch_start(self, batch: Any, batch_idx: int, dataloader_idx: int = 0) -> None:
```